app/lua/loadlib.c

```diff
@@ -480,7 +485,9 @@ static int ll_require (lua_State *L) {
     lua_pop(L, 1);
   }
   /* else must load it; iterate over available loaders */
-  lua_getfield(L, LUA_ENVIRONINDEX, "loaders");
+  lua_getfield(L, LUA_GLOBALSINDEX, "package");
```

Comment:
Oh this was a real doozy that took something for me to get my head around.  I still haven't worked out why the original version worked.  C closures can have an environment associated with then and in the case standard Lua of the init hook for a given library has the packages table as its environment. This also allow it to have a `__gc` meta entry.  This is all to support dynamic loaded libraries so that when the library is no longer used, then the GC can fire to unload it.  

None of this is relevant to our firmware where libraries are statically linked and referenced by a ROTable assembled during the link.  Also we use lightweight C closures which _can't_ have a separate environment associated.  The easiested work around  if just to reference the package table directly in _G.